src/timeline/index.html

```diff
@@ -84,42 +84,48 @@
 					<div class="clip_top">
-						<div tl-clip-menu class="clip_menu" ng-show="!enable_razor" ng-mousedown="showClipMenu(clip.id, $event)" tooltip-enable="!enable_razor" tooltip="{{clip.title}}" tooltip-placement="bottom" tooltip-popup-delay="400"></div>
+						<div tl-clip-menu class="clip_menu" ng-if="!enable_razor" ng-mousedown="showClipMenu(clip.id, $event)" tooltip-enable="!enable_razor" tooltip="{{clip.title}}" tooltip-placement="bottom" tooltip-popup-delay="400"></div>
 
 						<!-- CLIP EFFECTS -->
-						<div ng-show="!enable_razor" class="effect-container" id="effects_{{clip.id}}">
+						<div ng-if="!enable_razor" class="effect-container" id="effects_{{clip.id}}">
 							<div class="clip_effects" ng-repeat="effect in clip.effects" id="effect_{{effect.id}}" style="background-color: {{ getEffectColor(effect.type) }};" ng-click="selectEffect(effect.id); $event.stopPropagation();" ng-right-click="showEffectMenu(effect.id); $event.stopPropagation();" tooltip="{{effect.type}} ({{$index+1}}/{{clip.effects.length}})"  tooltip-placement="bottom" tooltip-popup-delay="400">
 								{{effect.type[0]}}
 							</div>
 						</div>
 
-						<!-- CLIP KEYFRAME POINTS -->
-						<div ng-repeat="(point, value) in getKeyframes(clip)" id="point_{{clip.id}}_{{point}}_{{$index}}" style="left: {{(((point - 1) / (project.fps.num / project.fps.den) - clip.start) * pixelsPerSecond)}}px;" class="point_region" ng-show="clip.selected "><div class="point_icon"></div></div>
-
 						<!-- CLIP LABEL -->
 						<div class="clip_label" ng-class="getClipLabelStyle(clip)" tooltip-enable="!enable_razor" tooltip="{{clip.title}}" tooltip-placement="bottom" tooltip-popup-delay="400">{{clip.title}}</div>
-						<br class="cleared">
+                        <br class="cleared">
 					</div>
 					<br class="cleared">
-					<div ng-show="!clip.show_audio" class="thumb-container">
-						<img class="thumb thumb-start" ng-show="getThumbPath(clip)" ng-src="{{ getThumbPath(clip) }}"/>
+
+					<div ng-if="!clip.show_audio" class="thumb-container">
+						<img class="thumb thumb-start" ng-if="getThumbPath(clip)" ng-src="{{ getThumbPath(clip) }}"/>
 					</div>
-					<div ng-show="clip.show_audio" class="audio-container">
+					<div ng-if="clip.show_audio" class="audio-container">
 						<canvas tl-audio height="46px" width="{{canvasMaxWidth((clip.end - clip.start) * pixelsPerSecond)}}px" class="audio"></canvas>
 					</div>
+
+                    <!-- CLIP KEYFRAME POINTS -->
+                    <div class="point_region" ng-if="clip.selected">
+                        <div ng-repeat="(point, value) in getKeyframes(clip)" ng-click="selectPoint(clip, point)" id="point_{{clip.id}}_{{point}}_{{$index}}" class="point point_{{ value }}" style="left: {{(((point - 1) / (project.fps.num / project.fps.den) - clip.start) * pixelsPerSecond)}}px;"></div>
```

Comment:
Just a style suggestion, to start: _please_ wrap insanely long lines. It's almost impossible to read the code on GitHub when lines are 300+ characters long.

```suggestion
                        <div ng-repeat="(point, value) in getKeyframes(clip)"
                          ng-click="selectPoint(clip, point)"
                          id="point_{{clip.id}}_{{point}}_{{$index}}"
                          class="point point_{{ value }}"
                          style="left: {{(((point - 1) / (project.fps.num / project.fps.den) - clip.start) * pixelsPerSecond)}}px;"></div>
```

Reply:
<s>(Because everything _is_ so wide, I shortened the variables to `p` and `val`. The code's right there, it's not like anyone will forget what they mean. They aren't ever visible outside of this single HTML tag's scope. Still, YMMV on that.)</s>
Well, I _thought_ I had. :confused:

Reply:
That double division is hard on the brain, too.

What if we inverted the FPS fraction and multiplied, instead? (Might be more efficient, though I bet the math library optimizes it that way anyway so it probably doesn't matter performance-wise.)

There's also a superfluous pair of outer parens in there:
```html
<div ... 
     style="left: {{((point - 1) * (project.fps.den / project.fps.num) - clip.start)
                    * pixelsPerSecond}}px;">
</div>
```

Reply:
Good suggestion, I'll wrap things up nicer